qemu/tests/balloon_check.py

```diff
@@ -148,6 +148,19 @@ def run_balloon_sub_test(self, test, params, env, test_tag):
             qemu_quit_after_test = 0
         return qemu_quit_after_test
 
+    def wait_for_balloon_complete(self):
+        """
+        Wait until guest memory don't change after do balloon
+        """
+        logging.info("Wait until guest memory don't change")
+        gmem = self.get_memory_status()
+        while True:
+            time.sleep(30)
+            cur_gmem = self.get_memory_status()
+            if abs(cur_gmem - gmem) < 100:
+                break
+            gmem = cur_gmem
```

Comment:
Well, it can be implemented with `utils_misc.wait_for()` in this way:
```python
def _mem_state():
    stable = False
    ori_mem = self.get_memory_status()
    while True:      
        yield stable
        cur_mem = self.get_memory_status()
        stable = abs(cur_mem - ori_mem) < 1
        ori_mem = cur_mem

is_stable = _mem_state()
utils_misc.wait_for(is_stable.next, timeout=timeout, step=10.0)
```

Also suggest shorter interval and small delta.